azure/scope/identity.go

```diff
@@ -185,17 +185,19 @@ func (p *AzureCredentialsProvider) GetClientID() string {
 // NOTE: this only works if the Identity references a Service Principal Client Secret.
 // If using another type of credentials, such a Certificate, we return an empty string.
 func (p *AzureCredentialsProvider) GetClientSecret(ctx context.Context) (string, error) {
-	secretRef := p.Identity.Spec.ClientSecret
-	key := types.NamespacedName{
-		Namespace: secretRef.Namespace,
-		Name:      secretRef.Name,
-	}
-	secret := &corev1.Secret{}
+	if secretRef := p.Identity.Spec.ClientSecret; secretRef.Name != "" {
```

Comment:
I assume this condition distinguishes between user-assigned and managed identity? If so would it be more expressive to put the criteria inside a convenience func, i.e., `isManagedIdentity()` (or `isUserAssignedIdentity()`, whichever is clearer)?

Reply:
This is meant to be only for indentities based on a SP id and password. Added a helper function to clarify.